src/plugin/datasource.ts

```diff
@@ -25,11 +30,47 @@ export function WavefrontDatasource(instanceSettings, $q, backendSrv, templateSr
 
     if (instanceSettings.jsonData.wavefrontToken) {
         this.requestConfigProto.headers["X-AUTH-TOKEN"] = instanceSettings.jsonData.wavefrontToken;
-    } else {
+    } else if (instanceSettings.jsonData.cspAPIToken) {
+        try {
+            fetch(CSP_API_TOKEN_URL, {
+                method: "POST",
+                body: JSON.stringify({
+                    "api_token": instanceSettings.jsonData.cspAPIToken,
+                }),
+                headers: {
+                    "Content-type": "application/x-www-form-urlencoded; charset=UTF-8"
+                }
+            })
+            .then((response) => response.json())
+            .then((json) => console.log(json));
+        } catch(e) {
+            console.error(e);
+        }
+        this.requestConfigProto.headers["Authorization"] = "Bearer " + instanceSettings.jsonData.cspAPIToken;
```

Comment:
I think we should set "Authorization" header to access token which we get after making call to CSP_OAUTH_TOKEN_URL. Not directly to csp api token. Please confirm this point from Gangadhar or Sadanand.
If they confirm, then change line#45 from : 
`.then((json) => console.log(json));`
to:
`then((accessToken) => this.requestConfigProto.headers["Authorization"] = accessToken)`

Reply:
Same with OAuth